core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpFunctions.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Provides various utility functions for internal use related with {@link DynamicHttpFunction}.
+ */
+final class DynamicHttpFunctions {
+
+    /**
+     * Returns {@link ResponseConverter} which can convert given {@code type} into {@link HttpResponse}
+     * from the configured converters.
+     *
+     * @throws IllegalArgumentException if no appropriate {@link ResponseConverter} exists for given
+     * {@code type}
+     */
+    private static ResponseConverter converter(Class<?> type, Map<Class<?>, ResponseConverter> converters) {
+        // search for the converter mapped to itself or one of its superclasses, except Object.class.
+        Class<?> current = type;
+        while (current != Object.class) {
+            ResponseConverter converter = converters.get(current);
+            if (converter != null) {
+                return converter;
+            }
+            current = current.getSuperclass();
+        }
+
+        // search for the converter mapped to one of its interface.
+        for (Class<?> itfc : type.getInterfaces()) {
```

Comment:
You also need to look for the interfaces of superclasses. See `ThriftServiceMetadata.getAllInterfaces()`